RATEST-242 : Patient Condition workflow test

mherman22 opened the pull request:
**Ticket ID**
https://issues.openmrs.org/browse/RATEST-242.

**Description**
The purpose of this pull request is to separate the conditions workflow from the [All Firefox Test](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/.github/workflows/qa.yml) by creating an independent conditions workflow and also fix the previously ignored scenarios in the conditions.feature file

**Reference**
https://talk.openmrs.org/t/creation-of-conditions-workflow-as-an-independent-workflow/34913

**Things done to that effect**
1. created conditions workflow
2. created the conditions Hooks i.e @conditions in the `RunTest.java` file
3. changed the hooks in the conditionsSteps to `SELENIUM_CONDITIONS` in the @after and @before methods.
4. created a script for it in the `package.json` file.
5. changed the tags in the conditions feature file from @dashboard to @conditions
6. created the workflow badge in the readMe file
6. Then fixed all the ignored scenarios in the conditions scenario

kdaud commented:
@mherman22 are you able to reproduce it locally?

mherman22 commented:
yes @kdaud i am able to reproduce it locally

mherman22 commented:
> Does the test pass when you comment out the last scenario?
> 
> something like
> 
> ```js
>   #@selenium
>   #@dashboard
>   Scenario: Delete first condition
>     And User clicks delete condition
>     Then System should trash first condition
> ```

yes it passes successfully as shown [here](https://pastebin.com/nmPfQVRh).

sorry for the late reply but this particular test hangs alot especially when changing from `TAB_ACTIVE`  to `TAB_INACTIVE` and when setting the conditions active or inactive

kdaud commented:
@mherman22 what happens if you change the element reference at line code [175](https://github.com/openmrs/openmrs-contrib-qaframework/blob/0b5737d2533b630df23212a6b37dfaf4dfde0c51/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L175) and [182](https://github.com/openmrs/openmrs-contrib-qaframework/blob/0b5737d2533b630df23212a6b37dfaf4dfde0c51/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L175) to

```js
driver.findElement(By.cssSelector("#remove-condition-dialog .confirm")).click();

```

mherman22 commented:
> Are you able to see the UI page element that is failing to be clickable? You need to change headless to false and observe the element during the test run.

yes it does, the dialog box is thrown when the test clicks the ❌ button but it just cant confirm the delete

mherman22 commented:
> is this PR updated as per the conversation here?

Am on it, just facing a bump trying to get [another branch](https://github.com/mherman22/openmrs-contrib-qaframework/tree/mherman22-patch-1) onto this pull request

mherman22 commented:
> have you just forgotten to add a workflow configuration that will trigger the condition test to run?

i created the workflow on this [other branch-mherman22-patch-1](https://github.com/mherman22/openmrs-contrib-qaframework/tree/mherman22-patch-1) but i have failed to figure out how to add that branch onto this pull request

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -113,8 +115,9 @@ public void setActive() {
 		conditionsPage.clickInActiveTab();
-		assertNotNull(conditionsPage.getFirstConditionName());
-		assertNotNull(getElement(ConditionsPage.SET_ACTIVE));
+		if (StringUtils.isNotBlank(conditionsPage.getFirstConditionName())) {
```
Why are you combining it in if statement, any reason for that

mherman22 replied:
the test fails to move the condition to inactive section `Results :

Failed tests:   Set first condition to inactive(Patient's Conditions Managment)` in the previous attempt using this 		```
assertNotNull(conditionsPage.getFirstConditionName());
		assertNotNull(getElement(ConditionsPage.SET_ACTIVE));
```
but when i use the if statement, it is moved to the inactive

mherman22 commented:
This is what i did since the last i posted here;

1. changed the conditions feature file as suggested by @kdaud so that the test workflow would be routing from the active visits, then patient dashboard and hence conditions page as [referenced here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/174#discussion_r735799611).

i have however faced snags in relationship to that approach. lets start here --> https://pastebin.com/unXc5ASy and the conditions file is https://pastebin.com/HfQLEKwj. as logs suggest, the first four tests run successfully however the next steps fail which are;- 

- `Set first condition to active`
- `Set first condition to inactive`
-` Edit first active condition`
- `Edit first inactive condition`
- `Delete first condition`

cc : @kdaud @sherrif10 @Irenyak1 @Muta-Jonathan @Ndacyayisenga-droid

kdaud commented:
@mherman22 update the PR with your changes to take a loom at the logs from the build plan of github actions

mherman22 commented:
@kdaud approval those runs / builds and we see how it behaves

kdaud commented:
done the needful !!

Muta-Jonathan commented:
@mherman22 
> Edit first active condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)
  Edit first inactive condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)
  Delete first condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)

Have a look 🆙 tests which are causing the Build Failure 
@kdaud Any hint why it is failing

kdaud commented:
> Any hint why it is failing

@Muta-Jonathan [cucumber report](https://reports.cucumber.io/reports/8628ce92-7042-472b-a2a6-4bf20d7d8c9f) tells the step(s) that is/are failing the plan hence need to be investigated and fixed

mherman22 commented:
> @mherman22
> 
> > Edit first active condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)
> > Edit first inactive condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)
> > Delete first condition(Patient's Conditions Managment): Error communicating with the remote browser. It may have died.(..)
> 
> Have a look up tests which are causing the Build Failure @kdaud Any hint why it is failing

thanks @Muta-Jonathan. i was able to see that through the cucumber reports

mherman22 commented:
> > Any hint why it is failing
> 
> @Muta-Jonathan [cucumber report](https://reports.cucumber.io/reports/8628ce92-7042-472b-a2a6-4bf20d7d8c9f) tells the step(s) that is/are failing the plan hence need to be investigated and fixed. cc: @mherman22

hey @kdaud i would have updated the PR already but im facing > openmrs-contrib-qaframework@0.1.0 refapp2Condition /home/mherman22/newForks/op - Pastebin.com as something seems off with how i have changed the feature file

kdaud commented:
@mherman22 share the content of RunTest file

mherman22 commented:
hey @kdaud i would have updated this PR already but im facing > openmrs-contrib-qaframework@0.1.0 refapp2Condition /home/mherman22/newForks/op - Pastebin.com as something seems off with how i have changed the feature file

> @mherman22 share the content of RunTest file

https://pastebin.com/qa7QAnxj

kdaud commented:
use `Scenario` key word instead of  `Scenario Outline`

kdaud commented:
@mherman22 according to the logs you [shared](https://pastebin.com/X25U1rse) it seems like you messed up with some configurations in RunTest file, I would recommend you delete your local repo and clone a new one from your fork

mherman22 commented:
> according to the logs you [shared](https://pastebin.com/X25U1rse) seems like you messed up with some configurations in RunTest file, I would recommend you delete your local repo and clone a new one from your fork

i did as advised here, however when i try to input the suggested changes of turning those scenarios into user stories, it fails again and in turn fails all the other tests

kdaud commented:
@mherman22 is your PR updated with your local changes?

mherman22 commented:
> @mherman22 is your PR updated with your local changes?

not yet

kdaud commented:
@mherman22 why not update your PR and reviewers take a look to point you where things might be failing?

mherman22 commented:
> 2 why not update your PR and reviewers take a look to point out where things might be failing from?

no new changes yet but as seen from this [cucumber report](https://reports.cucumber.io/reports/a9641231-d53c-4b7e-bdb2-832ab09d12f9), they should be able to guide.

kdaud commented:
are the changes here the ones you have locally on your branch?

mherman22 commented:
> are the changes here the ones you have locally on your branch?

yeah they are

mherman22 commented:
i was solving conflicts hence that commit

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -17,27 +20,32 @@ Feature: Patient's Conditions Managment
+  @condition
   Scenario: Delete first condition
     And User clicks delete condition
     Then System should trash first condition
```
isn't the feature file supposed to be

```js
Feature: Patient's Conditions Managment

  Background:
    Given User clicks on Conditions from Patient dashboard
    Given a user clicks on the active visits link from home page
    When a user selects a patient from the active patient list
    Then the system loads the Patient dashboard page
    When a user clicks on Conditions from Patient dashboard
    Then System loads Manage Conditions Page

  @selenium
  @condition
  Scenario Outline: Adding an active or inactive condition
    And User clicks on Add new condition
    Then System on Add New Condition Page
@@ -17,27 +20,32 @@ Feature: Patient's Conditions Managment
      | active   |
      | inactive |

 
     # User Story: Set first condition to inactive
    And User clicks on set inactive button
    Then System should move condition to inactive section

     #User Story: Set first condition to active
    And User clicks on set active button
    Then System should move condition to active section

     #User Story: Edit first active condition
    And User edits active
    Then System should edit all active adjustable fields

     #User Story: Edit first inactive condition
    And User edits inactive
    Then System should edit all inactive adjustable fields

    #User Story: Delete first condition
    And User clicks delete condition
    Then System should trash first condition

mherman22 replied:
looks okayy but when run it throws something similar to this -> https://pastebin.com/X25U1rse

kdaud replied:
doesn't the log come as a result of what was mentioned [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/174#issuecomment-958849299)?

mherman22 replied:
i did as you suggested, deleted my local repo, the cloned again. ran the test and it was throwing the initial errors/build failures we are trying to solve. However when i inserted those changes , it not-only fails but affects even other workflows

kdaud replied:
have you figured out the better approach to the workflow test?

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/RunTest.java:
```diff
@@ -41,14 +42,25 @@
+		public static final String SELENIUM_USER_ACCOUNT = SELENIUM + " and "
+				+ USER_ACCOUNT;
+		public static final String SELENIUM_REPORT = SELENIUM + " and "
+				+ REPORT;
```
you need to update your branch with changes from master!

mherman22 replied:
okay thanks

sherrif10 commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -39,37 +38,23 @@
     public void visitDashboard() {
         testPatient = createTestPatient();
         initiateWithLogin();
-        new TestData.TestVisit(testPatient.uuid, TestData.getAVisitType(),
```
Are we not using this method intatiation any where ?

mherman22 replied:
yes i decided to change direction and use the **find record link** instead of the visits link. i was hoping to get rid of the [snag here](https://reports.cucumber.io/reports/190a1576-922e-4959-ba58-eb92d3b1fa44) but unfortunately it doesn't.

please @sherrif10 look through and maybe you could tell me why the hook (@after) or `destroy()` method fail

mherman22 commented:
@kdaud trigger this please. lets see how it behaves here, 

it passes locally ---> https://pastebin.com/9waQvN07

mherman22 commented:
> @kdaud trigger this please. lets see how it behaves here,
> 
> it passes locally ---> https://pastebin.com/9waQvN07
@kdaud @sherrif10 how do i go about this?

kdaud commented:
@mherman22 the scenarios are failing because each of them is creating a different patient

```js
  @selenium
  @condition
  Scenario: Set first condition to inactive
    And User clicks on set inactive button
    Then System should move condition to inactive section

  @selenium
  @condition
  Scenario: Set first condition to active
    And User clicks on set active button
    Then System should move condition to active section

  @selenium
  @condition
  Scenario: Edit first active condition
    And User edits active
    Then System should edit all active adjustable fields

  @selenium
  @condition
  Scenario: Edit first inactive condition
    And User edits inactive
    Then System should edit all inactive adjustable fields

  @selenium
  @condition
  Scenario: Delete first condition
    And User clicks delete condition
    Then System should trash first condition
```

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -11,7 +11,7 @@ Feature: Patient's Conditions Managment
     And User enters "<activity>" condition
     And User clicks save
-    Then System loads Manage Conditions Page
+    Then System adds New Condition in Conditions table
```
Here are things that you may need to know;

1. Each scenario is independent from the other and when its run is done, the test patient is deleted from the db and when the test runs the next scenario it first creates its own test patient.
2. Scenarios that are failing is as a result of their steps missing adding condition in the condition's table so that the preceding steps can pass. Eg
```js
  @selenium
  @condition
  Scenario: Set first condition to inactive
    And User clicks on set inactive button
    Then System should move condition to inactive section
``
a use can can not click on set inactive button unless there a condition in the conditions table.

mherman22 replied:
> @mherman22 here are things that you may need to know;
> 
> 1. Each scenario is independent from the other and when its run is done, the test patient is deleted from the db and when the test runs the next scenario it first creates its own test patient.
> 2. Scenarios that are failing is as a result of their steps missing adding condition in the condition's table so that the preceding steps can pass. Eg
> 
> ```js
>   @selenium
>   @condition
>   Scenario: Set first condition to inactive
>     And User clicks on set inactive button
>     Then System should move condition to inactive section
> ```
> 
> set inactive button can only be visible on the Conditions page when there a condition in the conditions table.

thanks @kdaud for that clarity, i thought when i added `Then System adds New Condition in Conditions table` in the Scenario Outline i was catering for that.

Muta-Jonathan replied:
@kdaud  I think this might fail since [Qa-server is down](https://qa-refapp.openmrs.org/openmrs/)

Muta-Jonathan replied:
Maybe Re-run if it is up and running

mherman22 replied:
> I think this might fail since [Qa-server is down](https://qa-refapp.openmrs.org/openmrs/)

is it still down?, cuz it's running in my browser

mherman22 replied:
> Maybe Re-run if it is up and running

to add to this, am not quite sure why its failing against the server but running successfully locally on my pc

Muta-Jonathan replied:
It up and Running @mherman22 but am wondering Github actions seem not to run now on this PR ....

kdaud commented:
I am thinking of two options;

1. each scenario should create its condition for its steps to pass
2. refer to the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/174#discussion_r743144308)

mherman22 commented:
> 1. each scenario should create its own condition for its steps to pass.

this would mean separating the scenario outline into two independent scenarios i.e one for active condition and the other for inactive condition as shown below
```
  Background: 
    Given User clicks on Conditions from Patient dashboard
    Then System loads Manage Conditions Page

  @selenium @condition
  Scenario: Enter active condition
    When User clicks on Add new condition
    Then System on Add New Condition Page
    And User enters active condition
    And User clicks on save condition button
    Then System adds New Condition in Conditions table

  @selenium @condition
  Scenario: Enter inactive condition
    When User clicks on Add new condition
    Then System on Add New Condition Page
    And User enters inactive condition
    And User clicks on save condition button
    Then System adds New Condition in Conditions table
```

mherman22 commented:
cc: @kdaud @sherrif10 have your say

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/RunTest.java:
```diff
@@ -41,14 +42,25 @@
+		public static final String SELENIUM_FIND_PATIENT = SELENIUM + " and "
+				+ FIND_PATIENT;
+		public static final String SELENIUM_USER_ACCOUNT = SELENIUM + " and "
+				+ USER_ACCOUNT;
```
@mherman22 are you sure your branch is updated with changes from master?

mherman22 replied:
yes it is

kdaud replied:
checkout the code format in the [master](https://github.com/openmrs/openmrs-contrib-qaframework/blob/59743fa89c42ec7a054717d1ace933c70822f8f1/src/test/java/org/openmrs/contrib/qaframework/RunTest.java#L43-L52) and verify

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,65 @@
 
-  @dashboard
-  Scenario: Edit first active condition
+  @selenium @condition
```
any reason for putting the two tags on the same line?

mherman22 replied:
sorry about that, pretty formatter did that. lemi correct that

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -81,19 +87,25 @@ public void cancelAddNewCondition() {
+	}
+	
+	@Then ("System adds New Condition in Conditions table")
+	public void systemEntersExistingConditionToTable(){
```
how does it sound when the method  is renamed to systemAddsPatientCondition()?

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -171,33 +182,35 @@ public void successiveInactiveEdition() {
+	}
+		
+	@Then("System should trash first inactive condition")
+		public void SuccessfullyDeleteInctiveCondition() {
```
how about renaming the method to systemDeletesInActiveCondition()?

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java:
```diff
@@ -25,7 +25,8 @@
 	private static final By TAB_INACTIVE = By.cssSelector("a[href='#INACTIVE']");
 	private static final By FIRST_CONDITION_NAME = By.xpath("//table/tbody[2]/tr[1]/td[1]");
-	private static final By EDIT = By.cssSelector("i[title='Edit Condition: ']");
+	private static final By EDIT_ACTIVE_CONDITION = By.cssSelector("#ACTIVE tbody:nth-child(3) i.icon-pencil.edit-action.ng-scope");
```
is the element retrivable if you get rid of the nth-value?
```suggestion
	private static final By EDIT_ACTIVE_CONDITION = By.cssSelector("#ACTIVE i.icon-pencil.edit-action.ng-scope");
```

mherman22 replied:
> is the element retrievable if we get rid of the `nth-value`?

it shows 2 elements. that is as of my local instance. still waiting for the server to get back up and confirm

mherman22 replied:
`#ACTIVE tbody.ng-scope i.icon-pencil.edit-action.ng-scope` is the best option.

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/page/ConditionsPage.java:
```diff
@@ -25,7 +25,8 @@
 	private static final By FIRST_CONDITION_NAME = By.xpath("//table/tbody[2]/tr[1]/td[1]");
-	private static final By EDIT = By.cssSelector("i[title='Edit Condition: ']");
+	private static final By EDIT_ACTIVE_CONDITION = By.cssSelector("#ACTIVE tbody:nth-child(3) i.icon-pencil.edit-action.ng-scope");
+	private static final By EDIT_INACTIVE_CONDITION = By.cssSelector("#INACTIVE td:nth-child(4) i.icon-pencil.edit-action.ng-scope");
```
```suggestion
private static final By EDIT_INACTIVE_CONDITION = By.cssSelector("#INACTIVE i.icon-pencil.edit-action.ng-scope");
```

mherman22 replied:
`#INACTIVE tbody:nth-child(4)  i.icon-pencil.edit-action.ng-scope` that is the element that i  get from https://qa-refapp.openmrs.org/openmrs so for the your suggestion, it throws 2 elements (2/2)

mherman22 commented:
@kdaud @sherrif10

kdaud commented:
@mherman22 look into the merge conflicts and resolve!

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -8,14 +8,17 @@
 import org.apache.commons.lang3.StringUtils;
 import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.helper.TestData;
+import org.openmrs.contrib.qaframework.page.ClinicianFacingPatientDashboardPage;
```
do you need this import?
```suggestion
import org.openmrs.contrib.qaframework.page.ClinicianFacingPatientDashboardPage;
```

mherman22 replied:
it's used here -> https://github.com/openmrs/openmrs-contrib-qaframework/blob/71b4fbe3139b5ffb16e9710d0a9c7a9b7335bb76/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L43

kdaud commented on src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java:
```diff
@@ -8,14 +8,17 @@
+import org.openmrs.contrib.qaframework.page.ClinicianFacingPatientDashboardPage;
 import org.openmrs.contrib.qaframework.page.ConditionPage;
 import org.openmrs.contrib.qaframework.page.ConditionsPage;
+import org.openmrs.contrib.qaframework.page.FindPatientPage;
```
and this as well
```suggestion
import org.openmrs.contrib.qaframework.page.FindPatientPage;
```

mherman22 replied:
used here aswell https://github.com/openmrs/openmrs-contrib-qaframework/blob/71b4fbe3139b5ffb16e9710d0a9c7a9b7335bb76/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L41

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,69 @@
+  Scenario: Delete first active condition
+    And User enters active condition
+    And User clicks save
+    
```
get rid of the empty line here!

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,69 @@
+    # User story: Delete first active active condition
+    And User clicks delete first active condition
+    Then System should trash first active condition
+
```
even here
```suggestion
```

kdaud commented on src/test/resources/features/refapp-2.x/12-condition/conditions.feature:
```diff
@@ -1,43 +1,69 @@
+  Scenario: Delete first inactive condition
+    And User enters inactive condition
+    And User clicks save
+    
```
```suggestion
```

mherman22 commented:
@kdaud am gonna ahead and do the space deletion for the rest of the scenarios, is that okay?

kdaud commented:
> am gonna ahead and do the space deletion for the rest of the scenarios, is that okay?

@mherman22 would be great if you log a ticket and do the work there

mherman22 commented:
> would be great if you log a ticket and do the work there

elaborate more please

kdaud commented:
In response to the comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/174#discussion_r750173085), a child class can use the parents global variables by default. Now that Steps has declared [ClinicianFacingPatientDashboardPage](https://github.com/openmrs/openmrs-contrib-qaframework/blob/e7133bfefb08631f5e109de852db42e5bd8e7817/src/test/java/org/openmrs/contrib/qaframework/automation/Steps.java#L37) and [FindPatientPage](https://github.com/openmrs/openmrs-contrib-qaframework/blob/e7133bfefb08631f5e109de852db42e5bd8e7817/src/test/java/org/openmrs/contrib/qaframework/automation/Steps.java#L35) as its variables, can't `ConditionSteps` use them since its a child class to Steps?

kdaud commented:
> elaborate more please

create a ticket at [QA-Kanban](https://issues.openmrs.org/secure/RapidBoard.jspa?rapidView=240&selectedIssue=RESTWS-853) board  for that issue and resolve it!

mherman22 commented:
> > elaborate more please
> 
> create a ticket at [QA-Kanban](https://issues.openmrs.org/secure/RapidBoard.jspa?rapidView=240&selectedIssue=RESTWS-853) board for that issue and resolve it!

https://issues.openmrs.org/projects/RATEST/issues/RATEST-269?filter=addedrecently @kdaud  you make it ready for work

mherman22 commented:
> `ConditionSteps` use them since its a child class to Steps?

This is what is reflected there as shown 
```
        findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord().waitForPage();
        findPatientPage.enterPatient(testPatient.identifier);
        dashboardPage = (ClinicianFacingPatientDashboardPage) findPatientPage.clickOnFirstPatient().waitForPage();
```
But when applying the waitForPage() function, i need to do casting hence use of those pages in reference

kdaud commented:
@mherman22 if the ClinicianFacingPatientDashboardPage and FindPatientPage is declared by the super class, is there a need these variables to be declared again in child class?

mherman22 commented:
> if the ClinicianFacingPatientDashboardPage and FindPatientPage is declared by the super class, is there a need these variables to be declared again in child class?

no need

kdaud commented:
so why not get rid of those imports in the child class since its already defined in the super class?

mherman22 commented:
> so why not get rid of those imports in the child class since its already defined in the super class?

done that in this commit

kdaud commented:
> > > elaborate more please
> > 
> > 
> > create a ticket at [QA-Kanban](https://issues.openmrs.org/secure/RapidBoard.jspa?rapidView=240&selectedIssue=RESTWS-853) board for that issue and resolve it!
> 
> https://issues.openmrs.org/projects/RATEST/issues/RATEST-269?filter=addedrecently @kdaud am making you designated commiter with all due respect

@mherman22 checkout the comment on the ticket!

kdaud commented on .github/workflows/refapp-2.x-condition.yml:
```diff
@@ -0,0 +1,40 @@
+name: RefApp 2.x Condition
```
how does it sound to rename this to `RefApp 2.x Patient Condition`?

kdaud reviewed (approved):
Great work @mherman22 keep up the spirit of automation.

@sherrif10 do you have any more recommendation before we get this merged?

kdaud commented:
@mherman22 update this PR with changes from master.

sherrif10 commented:
@kdaud  was this oversight that i needed to first look through and perhaps approve before you triggered merge.

kdaud commented:
> @kdaud was this oversight that i needed to first look through

@sherrif10 after tagging you [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/174#pullrequestreview-807443225) and could not see your response after some 17 hours I thought that you should be occupied with other work elsewhere.

> perhaps approve before you triggered merge

this is always important for you to look through and approve!

kdaud commented:
@sherrif10 if there is any recommendation concerning the workflow, perhaps you can rise it and @mherman22 can do a follow up commit!

sherrif10 commented:
@kdaud  sometimes you need to hold on because you have been ephasising this on my prs and they used to revert them , so i think its better to send a reminder. All is well though.

kdaud commented:
> so i think its better to send a reminder

I agree with you!
